### ISSUE TITLE
[Structure]: Repository Structure

### ISSUE THREAD

**Issue by Maximus7474** [ManahilMuhammad/asp-team-20#13]
### Suggested Improvements
Currently the repository is set up in a way where all documentation is located in the root folder, to better organize the various files using a `Docs` directory and subdirectories to organize the data further is interesting.

### Reason for Change
As the development will start, the repository will contain more and more files that don't all relate to each other.
There will be the source code for the project along with the research data and installation/setup documentation for development.

### Additional Context
A possible structure for the repository could be the following:
```plaintext
.github/
research/
├── interview/
│   └── recordings
├── aims-objectives.md
└── competitor-analysis.md
documentation/
├── setup_guides/
│   ├── server.md
│   └── client.md
└── CONTRIBUTING.md
src/
├── server/
│   └── server source code
├── client/
│   └── client source code
├── dist/
│   └── (not commited) built project
├── package.json
└── .env
.gitattributes
.gitignore
README.md
LICENSE.md
```

**Comment by ManahilMuhammad** [comment]
I agree with this change, but I would suggest we put `aims-objectives.md` in a more suitable folder as it doesn't necessarily concern research. Or we could potentially delete the `aims-objectives.md` file because its content is already present in the `README.md` file.

**Comment by Maximus7474** [comment]
> Or we could potentially delete the `aims-objectives.md` file because its content is already present in the `README.md` file.

I'd agree more with this idea
